llvm/lib/Analysis/DependenceAnalysis.cpp

```diff
@@ -3649,6 +3646,111 @@ DependenceInfo::depends(Instruction *Src, Instruction *Dst) {
     LLVM_DEBUG(dbgs() << "can't analyze SCEV with different pointer base\n");
     return std::make_unique<Dependence>(Src, Dst);
   }
+
+  auto EltSize = SrcLoc.Size.toRaw();
+  assert(EltSize == DstLoc.Size.toRaw() && "Array element size differ");
+
+  // Check that memory access offsets in V are multiples of array EltSize.
```

Comment:
Please elaborate more. What is the assumed base ptr? What is is `Param`. Where does EltSize come from?